crates/trigger/src/lib.rs

```diff
@@ -113,18 +114,27 @@ impl<Executor: TriggerExecutor> TriggerExecutorBuilder<Executor> {
                 builder.add_host_component(outbound_redis::OutboundRedisComponent)?;
                 builder.add_host_component(outbound_pg::OutboundPg::default())?;
                 builder.add_host_component(outbound_mysql::OutboundMysql::default())?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    runtime_config::key_value::build_key_value_component(&runtime_config)?,
-                )?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    outbound_http::OutboundHttpComponent,
-                )?;
-                self.loader.add_dynamic_host_component(
-                    &mut builder,
-                    spin_config::ConfigHostComponent::new(runtime_config.config_providers()),
-                )?;
+                self.loader
+                    .add_dynamic_host_component(
+                        &mut builder,
+                        runtime_config::key_value::build_key_value_component(&runtime_config)?,
+                        init_data.kv,
```

Comment:
Unless I'm missing something you should be able to pass this init data into the function that creates the component here:

```suggestion
                        runtime_config::key_value::build_key_value_component(&runtime_config, init_data.kv)?,
```

and then in `build_key_value_component` do something like:

```rust
pub fn build_key_value_component(runtime_config: &RuntimeConfig, initial_values: Vec<(String, String)>) -> Result<KeyValueComponent> {
    let mut stores = runtime_config
        .key_value_stores()
        .context("Failed to build key-value component")?
        .collect::<HashMap<_, _>>();
    if !initial_values.is_empty() {
        if let Some(manager) = stores.get("default") {
            let store = manager.get("default").unwrap();
            for (key, value) in initial_values {
                store.set(&key, value.as_bytes())?;
            }
        } else {
            // bail? log warning?
        }
    }
    ...
```

Reply:
I was kind of thinking it would be nice to have the initialisation login in the component itself rather than as a side effect in a function in the `runtime_config` module.  But maybe I should get over myself!

Reply:
https://github.com/fermyon/spin/pull/1442/files/09bedc013db837176e3b63631b5e21f6a1688524#r1185541401